spec/Section 3 -- Type System.md

```diff
@@ -674,7 +980,66 @@ Interface types have the potential to be invalid if incorrectly defined.
    characters {"__"} (two underscores).
 
 
-### Unions
+### Interface Extensions
+
+InterfaceTypeExtension :
+  - extend interface Name Directives[Const]? FieldsDefinition
+  - extend interface Name Directives[Const]
+
+Interface type extensions are used to represent an interface which has been
+extended from some original interface. For example, this might be used to
+represent common local data on many types, or by a GraphQL service which is
+itself an extension of another GraphQL service.
+
+In this example, an extended data field is added to a `NamedEntity` type along
+with the types which implement it:
+
+```graphql example
+extend interface NamedEntity {
+  nickname: String
+}
+
+extend type Person {
+  nickname: String
+}
+
+extend type Business {
+  nickname: String
+}
+```
+
+Interface type extensions may choose not to add additional fields, instead only
+adding directives.
+
+In this example, a directive is added to a `NamedEntity` type without
+adding fields:
+
+```graphql example
+extend interface NamedEntity @addedDirective
+```
+
+**Type Validation**
+
+Interface type extensions have the potential to be invalid if incorrectly defined.
+
+1. The named type must already be defined and must be an Interface type.
+2. The fields of an Interface type extension must have unique names; no two
+   fields may share the same name.
+3. Any fields of an Interface type extension must not be already defined on the
+   original Interface type.
+4. Any Object type which implemented the original Interface type must also be a
+   super-set of the fields of the Interface type extension (which may be due to
+   Object type extension).
+5. Any directives provided must not already apply to the original Interface type.
+
+
+## Unions
+
+UnionTypeDefinition : Description? union Name Directives[Const]? UnionMemberTypes?
+
+UnionMemberTypes :
+  - = `|`? NamedType
+  - MemberTypes | NamedType
```

Comment:
I should be `- UnionMemberTypes | NamedType` as above, isn't it?

Reply:
Good catch. Looks like this bug only exists here and isn't replicated in the grammar summary